app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -862,9 +862,13 @@ public static ReadingData parseData(int attempt, String tagId, byte[] data, Long
                 historyList.add(glucoseData);
             }
         }
+        return historyList;
+    }
 
-
+    // Get the trend data for libre1/2
+    private static ArrayList<GlucoseData> parseTrendData(byte[] data, int sensorTime, long sensorStartTime, Long CaptureDateTime) {
```

Comment:
`parseHistoryData` and `parseTrendData` are almost identical with the exception of the index (26 vs 27) and data size (16 vs 32). To reduce code duplication these two functions could be merged.

Reply:
@tolot27: No: in the Libre Pro parsing the history data is a totally different matter: the historic data for the whole 14-day sensor life are stored in a contiguous memory space following the body and the two-byte pointer is stored at `78/79`. To read the FRAM blocks with an index higher than 255 you have to use the NFC custom commands `B0` (single block) or `B3` (maximum three blocks).

Reply:
@gui-dos I've just compared `parseHistoryData` and `parseTrendData` and the code is identical with the exception of the two mentioned cases. These can be parameterized. If the Libre Pro history ist totally different than something else should be implemented.